crates/core-new/src/compiler/mod.rs

```diff
@@ -0,0 +1,62 @@
+use std::collections::HashSet;
+
+use anyhow::Result;
+use hdp_primitives::{
+    datalake::task::DatalakeCompute,
+    module::Module,
+    task::{ExtendedTask, TaskEnvelope},
+};
+use hdp_provider::key::FetchKeyEnvelope;
+use module::ModuleCompilerConfig;
+
+pub mod datalake;
+pub mod module;
+
+pub struct Compiler {
```

Comment:
we could specify more traits in traits.rs
for Compiler and Processor and PreProcessor
not important rn

Reply:
compiler impl detail, can resolve later PR